locks/dynamodb/dynamo_lock.go

```diff
@@ -2,24 +2,52 @@ package dynamodb
 
 import (
 	"fmt"
-	"github.com/aws/aws-sdk-go/aws/session"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/defaults"
+	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/dynamodb"
-	"github.com/gruntwork-io/terragrunt/util"
 	"github.com/gruntwork-io/terragrunt/errors"
-	"github.com/aws/aws-sdk-go/aws"
+	"github.com/gruntwork-io/terragrunt/locks"
+	"github.com/gruntwork-io/terragrunt/util"
 )
 
 // A lock that uses AWS's DynamoDB to acquire and release locks
 type DynamoDbLock struct {
-	StateFileId 	string
-	AwsRegion   	string
-	TableName   	string
-	MaxLockRetries	int
+	StateFileId    string
+	AwsRegion      string
+	TableName      string
+	MaxLockRetries int
+}
+
+// New is the factory function for DynamoDbLock
+func New(conf map[string]string) (locks.Lock, error) {
+	if conf["state_file_id"] == "" {
```

Comment:
It's probably a good idea to extract all these string constants into `const` variables defined at the top of this file to keep things DRY.